### PR TITLE
Various Security Improvements feat. Nanako

### PR DESCRIPTION
**Pull request by RandomColonist**
This code is Nanako work, edited by me, mostly. Accidentally stumbled upon this branch and could not pass by.

Changes:

- Four new gun types, including: 

Wintermute, a high end military grade automatic rifle, designed for use against armoured infantry. Has a slightly slower firing rate than its contemporaries, but hits like a truck. Uses 7.62mm rounds.

Has 1.3 damage modifier, but fires only 250 RPM in full-auto mode.

image

Straylight, a compact and lightweight submachinegun that sprays small rounds rapidly. Sacrifices a fire selector to cut mass, so it requires a careful hand. Uses 9mm rounds.

Has 0.7 damage modifier, ONLY auto-fire mode with 600 RPM, can fit silencer. Can be VERY dangerous due to auto-only fire selector (auto-fire click catcher is really fucky, you will shoot if you try to even examine something or put gun on a table), so you better use safety when putting it down.

image

Paco, a modern and reliable sidearm for the soldier in the field. Uses 9mm rounds.
Has 1.1 damage modifier, can fit silencer.

image

Molly, an experimental fully automatic pistol. Compact and flexible, but somewhat underpowered. Uses 9mm rounds.

Has 0.55 damage modifier, which makes it very weak, but also has auto-mode with 600 RPM and fits in the holster, making it most compact full-auto SMG-pistol hybrid in game. It is very useful and can shred enemy upclose still, trust me. Can also fit silencer.

image

All these guns will be aviable in game one way or another, of course. They all added on Ultimate Protection disk too.

- ~~Almost all IH members (excluding Medical Specialist) will get random sidearm upon spawn. You can get either good old Martin with single cell for it, brand new Molly with one SMG mag or sturdy, reliable Paco with one pistol mag. Medical Specialist will always get Martin with one cell. Inspector will start with a random sidearm instead of second .44 revolver, go get it in your locker.~~ Cocknigger said that we are in army thus all operatives supposed to get same shit. Added handguns to vendors instead, everyone start with Martin and one cell.

- Silencer mechanics. Existing silencer for 9mm traitor handgun was repurposed by Nanako to fit on any gun if enabled. You can install/remove silencer by hand in a few seconds which produce screwdriwer-like sound and will make gun shots almost completely silent, new sound for silenced shot was also added by Nanako i believe. Silencer also will reduce gun damage by 10%. Guns that currently can fit silencer: Clarissa, Paco, Molly, Straylighter and one other gun - Mandella (top left on screenshot), which has an integrated silencer which can't be removed (integrated with new silencer mechanic).
image

- Fixed cases of missing caliber statement on a few guns, now you will always know what type of ammo do you need for any gun.

- Buffed Deckard damage by 10%.

- Slightly altered contents of IH lockers to make them not so cluttered. I moved all surplus equipment such as flashes, peppersprays, police tapes, and such into vendors, so lockers are not so badly cluttered now. I also planned to remove some spare armor from lockers, but it's kinda out of scope now. Also removed some spare equipment such as CO shield, go get one yourself in armory. Medspecs finally got medical AND tactical belt, stungloves and normal sidearm (Martin). Removed Bulls from Operative lockers  because they already have too much weapons and Bulls already present in armory.

- Security vendor contents changed drastically and now looks like that:
image

- General armory content was altered, changes are too many to list them, you can find out in game, now it looks like that:
image

- Secure armory changes:
Lightfall carbines was removed, instead four Straylights with ammo was added.
Slightly expanded secure armory by adding one more niche for ammo crate, which now contains ammo boxes for guns in armory and also disks for an autholathe.
Added two additional Bull shotguns.
Added two Wintermute heavy assault rifles with ammo.
image

- Firing range.
Replased all glass windows with walls, because what the fuck glass to fucking space doing on a fucking firing range?
image

### CODE REVIEW

**Comment by Voldirs** [comment]
I don't like lowering damage for pistol ammo again

**Comment by RandomColonist** [comment]
> I don't like lowering damage for pistol ammo again

Molly pistol uses 35 capasity SMG mags (thanks bug with ammo mags) by Nanako intention (you can see it by sprite). Yes, each shot is around ~11 damage, but it can shred targets with autofire in CQC, i kinda tested it. Yes, it's also ineffective against armor in general, but consider that you carry fucking SMG that can throw pretty accurate volley of bullets at enemies, in size of sidearm. I can rise damage to 0.6, but i kinda consider it pretty balanced right now, also i had to make it normal size because you could fit them in fucking pockets before. And imagine you aquired hollow-points or hight-velocities for this one. Sick DPS output.

Other pistols - i had to make Paco normal size intead of small because otherwise it would be just straight-up upgrade to traitor gun, which has all the same features, but deal 100% damage instead of 110%, so only holsters or storages now. And i also kinda buffed Deckard a little because it can use a little love and one bullet can make a difference sometimes.

**Comment by Voldirs** [comment]
Maybe leave ammo damage as it is and fumble with smg's damage modifier?

**Comment by RandomColonist** [comment]
> Maybe leave ammo damage as it is and fumble with smg's damage modifier?

Bring it up to cocknigger, i don't want to fumble witht those unless needed. Maybe i was unclear because of my english, but what i am trying to tell is a consider this pistol-smg hybrid pretty balanced because of size, ammo capacity and firing rate. It can shred things up close and cripple even.

If you mean auto-fire only SMG - it's like meatgrinder in a vaccum. If you will unleash 35 rounds up close ~14 damage each with 600RPM you will shred your target even faster. Again, test it yourself please, i consider them being pretty balanced at the moment.

**Comment by RandomColonist** [comment]
> Maybe leave ammo damage as it is and fumble with smg's damage modifier?

Sorry for inconvenience, i misunderstood you. Fixed.

**Comment by RandomColonist** [comment]
Finished work on autolathe implementation, tweaked map a little, made proper description. Ready to merge.

File: code/datums/outfits/outfit.dm [hunk 1]
```diff
@@ -189,3 +190,21 @@ var/list/outfits_decls_by_type_
 
 /decl/hierarchy/outfit/dd_SortValue()
 	return name
+
+//Some wrapper code to allow outfits to spawn random objects
+/proc/spawn_in_backpack(var/mob/M, var/newtype)
+	world << "Attempting backpack spawn [M] [newtype]"
```
Comment (ACCount12) [inline code/datums/outfits/outfit.dm]: REEEEEEEEEE

File: code/datums/outfits/outfit.dm [hunk 2]
```diff
@@ -189,3 +190,21 @@ var/list/outfits_decls_by_type_
...
+	var/list/atoms = list()
+	if (ispath(newtype, /obj/random))
+		var/obj/randomcatcher/R = new
+		//We make a randomcatcher and use it to get the results of the random thing
+		atoms = R.get_items(newtype)
+	else
+		atoms += new newtype
+
+	for (var/atom/movable/a in atoms)
+		world << "Moving and storing [a]"
```
Comment (ACCount12) [inline code/datums/outfits/outfit.dm]: REEEEEEEEEE

**Comment by RandomColonist** [comment]
> REEEEEEEEEEmove all the backpack changes, basically all the changes in `code/datums/outfits/jobs/security.dm`, and this is good to go.

Removed this and also random sidearms spawner, because we don't use it anywhere else.

**Comment by ACCount12** [comment]
gave it a kick to reset travis

**Pull request merged**